### PR TITLE
Add missing examples for std::cell types

### LINKED ISSUE #29344: API Docs: cell
Part of https://github.com/rust-lang/rust/issues/29329

http://doc.rust-lang.org/std/cell/

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #29344.

r? @steveklabnik

### CODE REVIEW

File: src/libcore/cell.rs [hunk 1]
```diff
@@ -237,6 +237,19 @@ impl<T:Copy> Cell<T> {
...
+    ///
+    /// ```
+    /// use std::cell::Cell;
+    ///
+    /// let mut c = Cell::new(5);
+    /// {
+    ///     let c2 = c.get_mut();
+    ///     *c2 += 1;
+    /// }
+    /// assert_eq!(c, Cell::new(6));
```
Comment (apasel422) [inline src/libcore/cell.rs]: Can this just be

```rust
assert_eq!(c.get(), 6);
```

Comment (GuillaumeGomez) [inline src/libcore/cell.rs]: Sure!

File: src/libcore/cell.rs [hunk 2]
```diff
@@ -237,6 +237,19 @@ impl<T:Copy> Cell<T> {
...
     /// This call borrows `Cell` mutably (at compile-time) which guarantees
     /// that we possess the only reference.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::cell::Cell;
+    ///
+    /// let mut c = Cell::new(5);
+    /// {
```
Comment (apasel422) [inline src/libcore/cell.rs]: Can this block just be replaced with

```rust
*c.get_mut() += 1;
```

File: src/libcore/cell.rs [hunk 3]
```diff
@@ -508,6 +548,19 @@ impl<T: ?Sized> RefCell<T> {
...
     /// This call borrows `RefCell` mutably (at compile-time) so there is no
     /// need for dynamic checks.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::cell::RefCell;
+    ///
+    /// let mut c = RefCell::new(5);
+    /// {
```
Comment (apasel422) [inline src/libcore/cell.rs]: Can this block just be replaced with

```rust
*c.get_mut() += 1;
```

File: src/libcore/cell.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline src/libcore/cell.rs]: Sure!

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**